solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -525,8 +652,14 @@ private void constructState(Set<String> changedCollections) {
     Map<String, ClusterState.CollectionRef> result = new LinkedHashMap<>();
 
     // Add collections
-    for (Map.Entry<String, DocCollection> entry : watchedCollectionStates.entrySet()) {
-      result.put(entry.getKey(), new ClusterState.CollectionRef(entry.getValue()));
+    for (Entry<String, StatefulCollectionWatch> entry :
+        collectionWatches.watchedCollectionEntries()) {
+      if (entry.getValue().currentState != null) {
+        // if the doc is null for the collection watch, then it should not be inserted into the
+        // state
+        result.putIfAbsent(
```

Comment:
`result` starts out as an empty map, so when you loop through the keys of another map and add them to `result`, there can't be duplicates. This could honestly just be ` put()`, I think.

Reply:
I'm scratch my head too 🤣 , as the latest code is not doing `putIfAbsent` here either. I feel like I could be just copying the original code from some earlier [commit](https://github.com/apache/solr/blame/54c5dd7d6d8b486df9132e637a6d7c5871ac4664/solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java#L591)  

Anyway, using `put` instead!